File: Task/OHalloran-numbers/Go/ohalloran-numbers.go
package main

import "fmt"

func main() {
    found := make([]bool, 1000) // all false initially
    for l := 1; l < 498; l++ {
        for w := 1; w <= l; w++ {
            lw := l * w
            if lw >= 498 {
                break
            }
            for h := 1; h <= w; h++ {
                sa := (lw + w*h + h*l) * 2
                if sa < 1000 {
                    found[sa] = true
                }
            }
        }
    }
    fmt.Printf("All known O'Halloran numbers:\n[")
    for i := 6; i < 1000; i += 2 {
        if !found[i] {
            fmt.Printf("%d, ", i)
        }
    }
    fmt.Printf("\b\b]\n")
}

File: Task/OHalloran-numbers/Nim/ohalloran-numbers.nim
import std/algorithm

const
  M = 1000      # Maximum area.
  N = M div 2   # Maximum half area.

var isOHalloran: array[3..N, bool]
isOHalloran.fill(true)

for length in 1..N:
  for width in 1..length:
    let plw = length * width
    if plw > N: break
    let slw = length + width
    for height in 1..width:
      let halfArea = plw + height * slw
      if halfArea > N: break
      isOHalloran[halfArea] = false

echo "All known O'Halloran numbers:"
for n in 3..N:
  if isOHalloran[n]:
    stdout.write 2 * n, ' '
echo()
